pkg/resources/kafka/pod.go

```diff
@@ -218,16 +217,17 @@ func getInitContainers(brokerConfig *v1beta1.BrokerConfig, kafkaClusterSpec v1be
 }
 
 func getVolumeMounts(brokerConfigVolumeMounts, dataVolumeMount []corev1.VolumeMount,
-	kafkaClusterName string, hasSSLSecrets bool) []corev1.VolumeMount {
+	listenersConfig *v1beta1.ListenersConfig, kafkaClusterName string) []corev1.VolumeMount {
 	volumeMounts := make([]corev1.VolumeMount, 0, len(brokerConfigVolumeMounts))
 	volumeMounts = append(volumeMounts, brokerConfigVolumeMounts...)
 
 	volumeMounts = append(volumeMounts, dataVolumeMount...)
 
-	if hasSSLSecrets {
-		volumeMounts = append(volumeMounts, generateVolumeMountForSSL()...)
+	if listenersConfig.SSLSecrets != nil || listenersConfig.ClientSSLCertSecretName != "" {
+		volumeMounts = append(volumeMounts, generateVolumeMountForClientSSLCerts())
 	}
 
+	volumeMounts = append(volumeMounts, generateVolumeMountForListenerCerts(listenersConfig)...)
```

Comment:
This should be guarded by an if. ListenerCerts are only needed when SSl is enabled.

Reply:
Than there will be a double check loop because we need to check SSL on every listener. 
In my solution generateVolumeMountForListenerCerts will return an empty array if there is no SSL on any listener and appending an empty array into an array equal the same array than before append.